### PR TITLE
fix(toolbar): hide feedback panel when submit feedback form opens

### LINKED ISSUE #74156: [DevToolbar] Hide toolbar modal when taking screenshot
Sentry Feedback: [JAVASCRIPT-2TVX](https://sentry.sentry.io/feedback/?referrer=github_integration&feedbackSlug=javascript%3A5592824094&project=11276)

When using the + button to send feedback, the toolbar's inbox still covers the screen when taking a screenshot.

|  |  |
| ------------- | --------------- |
| **contact_email** | andrew.liu@sentry.io |
| **name** | Andrew Liu |
| **source** | new_feedback_envelope |

### PR DESCRIPTION
**Pull request by aliu39**
Fixes https://github.com/getsentry/sentry/issues/74156 so the toolbar will be totally hidden when submitting feedback / taking a screenshot.

https://github.com/user-attachments/assets/6f733424-5520-495a-b06f-b17c6be50117

### CODE REVIEW

**Comment by michellewzhang** [comment]
the submit feedbac modal doesn't seem to show up when i try clicking +

https://github.com/user-attachments/assets/1c9ab2d2-1cb8-4155-80a0-4f2ed370ac9e

File: static/app/components/devtoolbar/hooks/useVisibility.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import type {CSSProperties, Dispatch, ReactNode, SetStateAction} from 'react';
+import {createContext, useContext, useState} from 'react';
+
+type State = CSSProperties['visibility'];
```
Comment (ryan953) [inline static/app/components/devtoolbar/hooks/useVisibility.tsx:4]: nice!

**Review by ryan953 — APPROVED** [review]

**Review by billyvg — APPROVED** [review]
The devtoolbar component being named `App` confused me a bit -- it made me thinking you were returning `null` for our main `App` view.

**Comment by aliu39** [comment]
Too late to change the title now but after the above recordings, this was changed to hide the entire toolbar (incl navigation bar) when the feedback widget is open

**Pull request merged**